models/result_test.go

```diff
@@ -470,6 +508,72 @@ var _ = Describe("Result", func() {
 		})
 	})
 
+	Describe("#Process", func() {
+		It("update result status to processing", func() {
+			user := FabricateUser(faker.Email(), faker.Password())
+			result := FabricateResult(user)
+
+			err := result.Process()
+			if err != nil {
+				Fail("Failed to process result")
+			}
+
+			Expect(result.Status).To(Equal(models.ResultStatusProcessing))
+		})
+
+		It("returns NO error", func() {
+			user := FabricateUser(faker.Email(), faker.Password())
+			result := FabricateResult(user)
+
+			err := result.Process()
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Describe("#Complete", func() {
+		It("update result status to completed", func() {
+			user := FabricateUser(faker.Email(), faker.Password())
+			result := FabricateResult(user)
+
+			err := result.Complete()
+			if err != nil {
+				Fail("Failed to compleete result")
+			}
+
+			Expect(result.Status).To(Equal(models.ResultStatusCompleted))
+		})
+
+		It("returns NO error", func() {
+			user := FabricateUser(faker.Email(), faker.Password())
+			result := FabricateResult(user)
+
+			err := result.Complete()
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Describe("#Fail", func() {
+		It("update result status to failed", func() {
```

Comment:
```suggestion
		It("updates result status to failed", func() {
```

Reply:
fixed in [538f129](https://github.com/carryall/go-google-scraper-challenge/pull/71/commits/538f12914d771bf610ef2f9a64ef5b41254eb52c)